scarlet/operator.py

```diff
@@ -248,7 +248,56 @@ def prox_soft_symmetry(X, step, strength=1):
     return X
 
 
-def prox_uncentered_symmetry(X, step, center=None, strength=.5, use_soft=True, fill=None):
+def prox_kspace_symmetry(X, step, shift=None, padding=10):
+    """Symmetry in Fourier Space
+
+    This algorithm by Nate Lust uses the fact that throwing
+    away the imaginary part in Fourier space leaves a symmetric
+    soution in real space. So `X` is transformed to Fourier space,
+    shifted by the fractional amount `shift=(dy, dx)`,
+    the imaginary part is discarded, shited back to its original position,
+    then transformed back to real space.
+    """
+    # Record the morph shape
+    shape = X.shape
+    dy, dx = shift
+    padding = np.max(X.shape) + padding // 2
+    edges = ((padding, padding), (padding, padding))
+    corner = (padding, padding)
+    zeroMask = X <= 0
+    X[zeroMask] = 0
```

Comment:
Zero masking before the FFTs should go.

Reply:
Right, but we still want to save the mask to reassign the zeros at the end, right?

Reply:
I think so. Otherwise, as @pmelchior said, the artifacts would mess with the threshold estimate when applying the l0. Although a preferable but still not ideal solution would probably be a smooth decrease rather than a sharp drop to zero as Nate pointed out.

Reply:
Either way the mask is useful anyway.

Reply:
fixed